libraries/bezier-rs/src/subpath/transform.rs

```diff
@@ -75,13 +86,184 @@ impl Subpath {
 			(Subpath::new(first_split, false), Some(Subpath::new(second_split, false)))
 		}
 	}
+
+	/// Returns [ManipulatorGroup]s with a reversed winding order.
+	fn reverse_manipulator_groups(manipulator_groups: &[ManipulatorGroup]) -> Vec<ManipulatorGroup> {
+		manipulator_groups
+			.iter()
+			.rev()
+			.map(|group| ManipulatorGroup {
+				anchor: group.anchor,
+				in_handle: group.out_handle,
+				out_handle: group.in_handle,
+			})
+			.collect::<Vec<ManipulatorGroup>>()
+	}
+
+	/// Returns a [Subpath] with a reversed winding order.
+	pub fn reverse(&self) -> Subpath {
+		Subpath {
+			manipulator_groups: Subpath::reverse_manipulator_groups(&self.manipulator_groups),
+			closed: self.closed,
+		}
+	}
+
+	/// Returns an open [Subpath] that results from trimming the original Subpath between the points corresponding to `t1` and `t2`.
+	/// If `t1` > `t2`, then behavior for computing the resulting Subpath will differ depending on whether the original Subpath is open or closed:
+	/// - If the original Subpath was open, an open subpath from `t1` to `t2` will be returned (with a winding order reverse that of the original Subpath).
+	/// - If the original Subpath was closed, an open subpath from `t1` to `t2` that crosses the break between 1 and 0 will be returned (winding in the same direction as the original Subpath from `t1` to `t = 1 = 0` to `t2`).
```

Comment:
Upon reading your tests, I see you wisely distinguished between closed subpaths which have their start and end points at the same, and different, locations. I hadn't considered that when I wrote this `t = 1 = 0` part of this comment, which I assume may be incorrect due to that fact. Please update this for me so it's correct again, assuming I did actually state an inaccuracy. Thanks.

Reply:
I mean you have tests for closed subpaths where the start and end points are located in the same position. And others where they are located in different positions (but it's still closed).

Reply:
Which tests are you referring to?

Reply:
Never mind, maybe I misread the tests. But even if I misread them, they still reminded me of the two cases where a closed subpath might, or might not, have endpoints at the same position. Please make sure you handle/account for that (and have tests for it). See https://files.keavon.com/-/SandyFancyYak/capture.png (note: each row has identical geometry on each side, with only the closed flag changed per side).

Reply:
I want to make sure we're on the same page about the behavior for these cases.

In our code, a subpath has an associated list of manipulator groups. When closing one, we simply assume that there is an additional bezier between the first and last elements of the manipulator group list.

In the first row of your above image, the open subpath has a start and end point at the exact same location. If the closed flag is enabled for this subpath, our code will keep the "duplicate" manipulator group at the end of the list, and treat it as if there's a bezier that travels between them. I.e. there's an extra bezier which is essentially a single point.

If you wanted the closed code to not have that additional bezier, the user would have to manually remove the extra manipulator group at the end.

Is this what you would expect?
Should we add a `close_subpath` function that will both enable the `closed` flag and remove the extra manipulator group if necessary?

Reply:
I confused myself, thank you for breaking this down to clarify my mental misconceptions. Everything you described is correct. Your `close_subpath` idea would indeed be a useful function doing as you described. So with that in mind, I think we can disregard anything I asked about in this whole code review comment's thread (but just confirm the doc comment is accurate after my possibly flawed changes). Thanks.

Reply:
I confused myself, thank you for breaking this down to clarify my mental misconceptions. Everything you described is correct. Your `close_subpath` idea would indeed be a useful function doing as you described. So with that in mind, I think we can disregard anything I asked about in this whole code review comment's thread (but just confirm the doc comment is accurate after my possibly flawed changes). Thanks.

Reply:
I confused myself, thank you for breaking this down to clarify my mental misconceptions. Everything you described is correct. Your `close_subpath` idea would indeed be a useful function doing as you described. So with that in mind, I think we can disregard anything I asked about in this whole code review comment's thread (but just confirm the doc comment is accurate after my possibly flawed changes). Thanks.